impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Funnel.js

```diff
@@ -0,0 +1,203 @@
+  "pentaho/visual/util"
+], function(module, BaseView, echarts, util) {
+
+  "use strict";
+
+  return BaseView.extend(module.id, {
+
+    _getEChartsLabel: function(labelsOption) {
+      switch(labelsOption) {
+        case "topLeft" :
+          return "leftTop";
+        case "bottomLeft" :
+          return "leftBottom";
+        case "topRight":
+          return "rightTop";
+        case "bottomRight":
+          return "rightBottom";
+        default:
+          return labelsOption;
+      }
+    },
+
+    _configureLabel: function(options) {
+      var font = util.getDefaultFont(null, 12);
+
+      var label = {
+        show: true,
+        position: this._getEChartsLabel(this.model.labelsOption),
+        formatter: "{b}: {d}%",
+        backgroundColor: "transparent",
+        fontSize: font.substring(0, font.indexOf(" ")),
+        fontFamily: font.substring(font.indexOf(" ") + 1)
+      };
+
+      options.series.forEach(function(row) {
+        row.label = label;
+      });
+    },
+
+    _configureLegend: function(options, records) {
+      var categories = [];
+      var font = util.getDefaultFont(null, 14);
+
+      records.forEach(function(record) {
+        categories.push(record.name);
+      }
+      );
+
+      options.legend = {
+        data: categories,
+        type: "scroll",
+        align: "auto",
+        padding: 10,
+        textStyle: {
+          fontSize: font.substring(0, font.indexOf(" ")),
+          fontFamily: font.substring(font.indexOf(" ") + 1)
+        }
+      };
+    },
+
+    _buildColors: function() {
+      return this.model.palette.colors.toArray(function(color) {
+        return color.value;
+      });
+    },
+
+    _buildTooltip: function(tooltipFormatString, font) {
+      return {
+        formatter: tooltipFormatString,
+        fontSize: font.substring(0, font.indexOf(" ")),
+        fontFamily: font.substring(font.indexOf(" ") + 1)
+      };
+    },
+
+    _buildSeries: function() {
+      var dataTable = this.model.data;
+      var range = dataTable.getColumnRange(dataTable.getNumberOfColumns() - 1);
+
+      return [
+        {
+          type: "funnel",
+          left: "20%",
+          right: "20%",
+          top: "10%",
+          bottom: "10%",
+          width: "auto",
+          height: "auto",
+          min: range.min,
+          max: range.max,
+          minSize: 5,
+          maxSize: "100%",
+          sort: this.model.order === "bySizeDescending" ? "descending" : "ascending",
+          legendHoverLink: true,
+          gap: 4,
+          labelLine: {
+            length: 10,
+            lineStyle: {
+              width: 1,
+              type: "solid"
+            }
+          },
+          itemStyle: {
+            borderColor: "#fff",
+            borderWidth: 1
+          },
+          emphasis: {
+            label: {
+              fontSize: 20
+            }
+          }
+        }
+      ];
+    },
+
+    _getTooltipFormatter: function(dataTable, rowNum) {
```

Comment:
It is the string format of the Tooltip.

Reply:
You mean, like an HTML template string? Because it seems like a _final_ HTML string, in which case the word format would not apply as well --- a format in this context typically designates a string which will contains placeholders, for example, that will still be processed somehow to build a formatted text. Maybe you mean "formatted"? Still, seems kind of redundant in this context, given a tooltip necessarily represents formatted text.

Reply:
Yes. It is a formatted string. 

I refactored it to what you suggested above -_buildRowTooltipHtml and rowTooltipHtml